pandas/core/reshape/tile.py

```diff
@@ -413,7 +432,10 @@ def _bins_to_cuts(
                 )
 
         if not is_categorical_dtype(labels):
```

Comment:
Do we want to add an `elif` branch that handles converting `Categorical` labels that don't have matching `ordered`, e.g. `labels=Categorical(["A", "B", "C"], ordered=False)` with `ordered=True` being specified in `cut`?

Not sure if we should actually do that or not, so fine to save this for a follow-up if more discussion is needed. Especially since it would be breaking behavior that we should probably warn users about first, making the implementation a bit messy as it'd likely require using a sentinel for `ordered`.

Reply:
My first instinct would be to raise for that particular case, but also open to thoughts.

Might be worthwhile putting a `#TODO` comment here noting that case.

Reply:
We could add a TODO and in the meantime raise a warning.

Reply:
I don't think a print message or a warning is completely necessary yet. Just a code comment should be good